### ISSUE TITLE
Create all entities by default, but disable those not in use.

### ISSUE THREAD

**Issue by TheDukeSr** [rroller/dahua#38]
**problem:**
Maybe I'm wrong, but the only way to enable a new entitie like Crosswire detection now is to remove and re-add the camera?

**suggestion:**
Why not create all available entities by default but Disable those the user does not want when cam is installed.
Then later we can just Enable or Disable the entities if we want to use them.

 [same for streams. see request #36 ](https://github.com/rroller/dahua/issues/36)

If #36 is also a good option to create, then this integration might even be easier to maintain or debug (no more different entities created depending the cam type, but allways the same entity list is created, just disabled what not needed).

**Comment by rroller** [comment]
Yeah that's an option. I'm also going to update the integration so you can reconfigure it and change things like usernames, passwords, address, etc. It's on my TODO.

**Comment by JSEHV** [comment]
Any change this will be soon implemented? 
I am really looking forward to these features!

**Comment by rroller** [comment]
All streams are now created by default.

**Comment by zios0007** [comment]
@TheDukeSr you were right, and the first half is fixed.

> the only way to enable a new entitie like Crosswire detection now is to remove and re-add the camera?

Not any more. #601 (**0.9.85**) made the event subscription editable after setup:

`Settings` -> `Devices & services` -> `Dahua` -> `Configure` -> *Events to subscribe to*

Change it, and the entry reloads with the new set. No removing and re-adding.

On the second half, creating every entity and letting the user disable what they do not want, the project went the other way and it is worth saying why, since your suggestion is the more obvious one.

Entities are created based on what the device is **probed** to support, not on a fixed list. The reason is that an entity which cannot work is worse than a missing one: it sits at `unavailable` or `unknown` forever, appears in automations and dashboards, and generates bug reports from people who reasonably assume it should do something. #679 removed a set of those, and #743 removed an Illuminator entity that was driving an infrared LED on cameras with no white light.

There is also a middle ground now that did not exist in 2021: platform toggles. `Configure` lets you switch off `select`, `light`, `switch` and the rest per device, which stops the polling those entities cause as well as hiding them.

@rroller this can be closed.
